config_utilities.hpp

```diff
@@ -905,13 +963,83 @@ struct ConfigInternal : public ConfigInternalVerificator {
   std::string rosParamNameSpace() {
     // Check scope and param map are valid.
     if (!meta_data_->params) {
-      LOG(WARNING) << "'rosParamNameSpace()' calls are only allowed within the "
-                      "'fromRosParam()' method, no param will be loaded.";
-      return "";
+      if (!meta_data_->merged_setup_currently_active) {
+        LOG(WARNING)
+            << "'rosParamNameSpace()' calls are only allowed within the "
+               "'fromRosParam()' method, no param will be loaded.";
+      }
+      return param_namespace_;
+    }
+    return param_namespace_;
+  }
+
+  void setupParam(const std::string& name, int* param) {
+    this->setupParamInternal(name, param);
+  }
+
+  void setupParam(const std::string& name, float* param) {
+    this->setupParamInternal(name, param);
+  }
+
+  void setupParam(const std::string& name, double* param) {
+    this->setupParamInternal(name, param);
+  }
+
+  void setupParam(const std::string& name, bool* param) {
+    this->setupParamInternal(name, param);
+  }
+
+  void setupParam(const std::string& name, std::string* param) {
+    this->setupParamInternal(name, param);
+  }
+
+  void setupParam(const std::string& name, std::vector<int>* param) {
+    this->setupParamInternal(name, param);
+  }
+
+  void setupParam(const std::string& name, std::vector<double>* param) {
+    this->setupParamInternal(name, param);
+  }
+
+  void setupParam(const std::string& name, std::vector<float>* param) {
+    this->setupParamInternal(name, param);
+  }
+
+  void setupParam(const std::string& name, std::vector<bool>* param) {
+    this->setupParamInternal(name, param);
+  }
+
+  void setupParam(const std::string& name, std::vector<std::string>* param) {
+    this->setupParamInternal(name, param);
+  }
+
+  void setupParam(const std::string& name, std::map<std::string, int>* param) {
+    this->setupParamInternal(name, param);
+  }
+
+  void setupParam(const std::string& name, std::map<std::string, double>* param) {
+    this->setupParamInternal(name, param);
+  }
+
+  void setupParam(const std::string& name, std::map<std::string, float>* param) {
+    this->setupParamInternal(name, param);
+  }
+
+  void setupParam(const std::string& name, std::map<std::string, bool>* param) {
+    this->setupParamInternal(name, param);
+  }
+
+  void setupParam(const std::string& name,
+                std::map<std::string, std::string>* param) {
+    this->setupParamInternal(name, param);
+  }
```

Comment:
Not a technical but a user friendliness reason (It used to be one templated function): The ros params are only defined for these types on the ROS side. By explicitly overloading it completion will instantly warn you that you do something strange or do implicit casts for you. Otherwise you get a really ugly compile error if an unknown argument is deducted which can be hard to understand and debug, in particular if this is used as 3rd party import.